akka-stream/src/main/boilerplate/akka/stream/scaladsl/GraphApply.scala.template

```diff
@@ -22,6 +22,9 @@ trait GraphApply {
    * Creates a new [[Graph]] by importing the given graph `g1` and passing its [[Shape]]
    * along with the [[GraphDSL.Builder]] to the given create function.
    */
+   /* FIXME stuck on Scala 3 not inferring Graph.Shape for g1.Shape to the concrete type here, while 2.13 does, so the graph builder
+    * implicits for the concrete type is not found and all GraphDSL usage fails to compile.
+    */
```

Comment:
Graph is 
```scala
trait Graph[+S <: Shape, +M] {
  type Shape = S @uncheckedVariance
  ...
}
```

Calling this ins Scala 2.13 that means that `g1.Shape` is inferred to the actual concrete type, for example `SourceShape`, and when the user does something like this:

```scala
RunnableGraph.fromGraph(Source.single("one")) { implicit b => s =>
  import GraphDSL.Implicits._
 ...
}
```

`s` is inferred to `SourceShape` which allows us to decorate it with various concrete `Shape`-subclass specific methods for building a graph.

In Scala 3 it however is inferred to `Shape @ uncheckedVariance` rather than the concrete type so the compiler cannot find any of those implicits.

Reply:
The real problem turns out to be the covariance of `S` and then also in the various concrete shape types in and out type parameters. For `GraphApply` we can get around it by introducing more type parameters for the graph shape types and use those instead of the path dependent type, however the `Shape` hierarchy seems trickier.

Reply:
Giving up for now, variance of the graph/stream APIs is really tricky, haven't figured out a strategy for how to deal with that. Existing APIs are sprinkled with `@uncheckedVariance` as a bail out for when we _know_ the types align but the compiler doesn't. Seems really hard to maintain source compatibiity of those classes for both Scala 2.13 and 3.0 even if we introduce version specific source trees.